common/XenomaiLock.cpp

```diff
@@ -49,28 +50,29 @@ static void initialize_xenomai() {
     xenomai_init(&argc, argvPtrs);
 }
 
-static int turn_into_cobalt_thread(bool recurred = false) {
-    int current_mode = cobalt_thread_mode();
```

Comment:
technically this variable is unused, and cobalt_thread_mode() seems to be a plain getter with no side effects, so it should be safe to remove, but i'm not certain that it's right to do so. perhaps this has necessary side effects, or the variable was intended to be used somewhere below.

Reply:
you are right, it's fine to remove it. I guess the code is leftover from a previous attempt. `cobalt_thread_mode()` would be an easier way to know whether we are a Xenomai thread, however it's not part of the stable API so I preferred not to use it, but I must've forgotten it in there.

Reply:
I am used to using `clang` who kindly warns me by default about unused variables ... maybe such a warning could be added as a default to Sc's build flags?

Reply:
> I am used to using clang who kindly warns me by default about unused variables ... maybe such a warning could be added as a default to Sc's build flags?

yes, i am all for that. this would tie into https://github.com/supercollider/supercollider/issues/2927, which i've actually just gone and added to our milestone for 3.13.